src/common/core/corediff.js

```diff
@@ -2511,6 +2518,9 @@ define(['common/util/canon',
         //</editor-fold>
 
         //<editor-fold=Added Methods>
+
+        // FIXME: It really looks like the diff requires that no nodes are mutated. This must be documented somewhere.
+        // FIXME: Maybe checking for isMutated in core.js at the roots are enough..
```

Comment:
yeah, that check should suffice
on the other hand, it is not completely true that you need to have 'not mutated' trees to compare. All the data is interpreted, so it should be possible with any kind of trees - though we still use the hashes of children to figure out if we should go deeper, but that could be changed...